apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -124,8 +124,54 @@ type SubmarinerList struct {
 	Items           []Submariner `json:"items"`
 }
 
+// BrokerSpec defines the desired state of Broker
+// +k8s:openapi-gen=true
+type BrokerSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	Components                  []string `json:"components,omitempty"`
+	DefaultCustomDomains        []string `json:"defaultCustomDomains,omitempty"`
+	GlobalnetCidrRange          string   `json:"globalnetCidrRange,omitempty"`
+	DefaultGlobalnetClusterSize uint     `json:"defaultGlobalnetClusterSize,omitempty"`
+	GlobalnetEnabled            bool     `json:"globalnetEnabled,omitempty"`
```

Comment:
We might require changes to subctl to prompt the user to provide the GlobalnetCIDR/GlobalnetClusterSize while joining a new Cluster, if there are no defaults configured when Broker is deployed.

Reply:
This is already the case, I’ll do that in a separate PR. https://github.com/submariner-io/submariner-operator/issues/1103 tracks this.

Reply:
I suggest we do something like:

https://github.com/submariner-io/submariner-operator/blob/devel/apis/submariner/v1alpha1/submariner_types.go#L131

https://github.com/submariner-io/submariner-operator/blob/devel/apis/submariner/v1alpha1/servicediscovery_types.go#L84

Reply:
Well, or may be not, not sure, if they are going through the process of using the operator directly.... may be it's better to let them be explicit

Reply:
I prefer https://github.com/submariner-io/submariner-operator/pull/1104 ;-)